apis/Google.Cloud.Bigtable.V2/Google.Cloud.Bigtable.V2/ColumnQualifier.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright 2017 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using Google.Protobuf;
+using System;
+
+namespace Google.Cloud.Bigtable.V2
+{
+    /// <summary>
+    /// A column qualifier.
+    /// </summary>
+    public sealed class ColumnQualifier : IEquatable<ColumnQualifier>
+    {
+        /// <summary>
+        /// Creates a new <see cref="ColumnQualifier"/> instance from the UTF8 encoding of a string.
+        /// </summary>
+        /// <param name="qualifierAsUtf8">The qualifier whose UTF8 representation should be used to initialize the qualifier.</param>
+        public ColumnQualifier(string qualifierAsUtf8)
```

Comment:
I figure since we have the implicit operators, those will be used more than either constructors or factory methods, so I'm not sure it matters too much which way we go. I don't feel too strongly about it. I guess one benefit to factory methods is we could then have methods to take a string as UTF8 and another as Base64. But then again, I don't want to duplicate the whole `ByteString` API. This is just meant as a handy wrapper with implicit operators so you can specify column qualifiers as strings if necessary. I'm leaning towards leaving as is.